itou/approvals/models.py

```diff
@@ -267,6 +267,34 @@ def can_be_suspended_by_siae(self, siae):
             and not self.user.last_accepted_job_application.can_be_cancelled
         )
 
+    @cached_property
+    def last_in_progress_suspension(self):
+        """
+        Returns the last in progress suspension if exists else None
+        """
+        if self.is_suspended:
+            return self.suspensions_by_start_date_asc.last()
+
+    @cached_property
+    def is_open_to_application_process(self):
+        if self.is_suspended:
+            return self.last_in_progress_suspension.reason in [
+                Suspension.Reason.BROKEN_CONTRACT.value,
+                Suspension.Reason.FINISHED_CONTRACT.value,
+            ]
+        return True
+
+    def update_last_suspension(self, hiring_start_at):
+        """
+        An SIAE can reduce the delay of a suspension when they hired job seeker.
+        So, if approval is open to application process and have an active suspension,
+        We save the end date to the d-1 hiring start.
+        """
+        active_suspension = self.last_in_progress_suspension
+        if self.is_open_to_application_process and active_suspension:
```

Comment:
```suggestion
        if active_suspension and self.is_open_to_application_process:
```

Ici tu peux inverser les clauses de la condition de façon à ne pas déclencher la vériication de la seconde si la première est fausse de façon à économiser des requêtes.

Reply:
bien vu :)